src/wp-includes/formatting.php

```diff
@@ -3358,6 +3358,12 @@ function convert_smilies( $text ) {
 	if ( get_option( 'use_smilies' ) && ! empty( $wp_smiliessearch ) ) {
 		// HTML loop taken from texturize function, could possible be consolidated.
 		$textarr = preg_split( '/(<.*>)/U', $text, -1, PREG_SPLIT_DELIM_CAPTURE ); // Capture the tags as well as in between.
+
+		if ( preg_last_error() != PREG_NO_ERROR ) {
+			trigger_error( preg_last_error_msg(), E_USER_WARNING );
```

Comment:
Instead of using trigger_error, `wp_trigger_error` should be used. First pass at a more informative error message.

```suggestion
			$message = sprintf( '<code>convert_smilies</code> could not split text. Error Message: %s', preg_last_error_message() );
			wp_trigger_error( 'convert_smilies', $message, E_USER_WARNING );
```